discover/types.go

```diff
@@ -183,3 +183,18 @@ func (si SystemInfo) GetOptimalThreadCount() int {
 
 	return coreCount
 }
+
+// Return if the system supports flash attention
+func (l GpuInfoList) SupportsFlashAttention() bool {
+	for _, gpu := range l {
+		// For each GPU, check if it does NOT support flash attention
+		supportsFA := gpu.Library == "metal" ||
+			(gpu.Library == "cuda" && gpu.DriverMajor >= 7) ||
+			gpu.Library == "rocm"
+
+		if !supportsFA {
+			return false // Found a GPU that doesn't support it
+		}
+	}
+	return true // All GPUs support it
```

Comment:
I'd put these in a GoDoc above:

```
// FlashAttentionSupported returns true if all GPUs support flash attention
```


```suggestion
			return false
		}
	}
	return true
```